### PR TITLE
In setupRecurrenceWithMasterEntit a null recurrence pattern is now inter...

### PR DESCRIPTION
**Pull request by javierag**
...preted as remove the current recurrence pattern.

Before this change it triggered a crash.

### CODE REVIEW

**Comment by javierag** [comment]
I put this  PR mainly for asking help. How can I could test this is correct?.

**Comment by sixstone-qq** [comment]
Typos in the commit message and not following the guidelines.

Do not forge to add as a prefix of the commit message: oc-calendar

File: OpenChange/MAPIStoreRecurrenceUtils.m [hunk 1]
```diff
@@ -65,6 +65,11 @@ - (void) setupRecurrenceWithMasterEntity: (iCalRepeatableEntityObject *) entity
   [entity removeAllExceptionRules];
   [entity removeAllExceptionDates];
 
+  if (rp == NULL) {
+    /* No recurrence pattern to set, so we return after removing the old ones */
+    return;
```
Comment (sixstone-qq) [inline OpenChange/MAPIStoreRecurrenceUtils.m]: If that is case at any time. It requires to have a log line indicating this strange pattern using `[self logWithFormat:]` or `NSLog` as fallback.

**Comment by blaxter** [comment]
On iCalEvent+MAPIStore.m _setupEventRecurrence function if `get_AppointmentRecurrencePattern` fails `pattern` will be NULL and then it will crash when trying to call:

    [(iCalCalendar *) parent
        setupRecurrenceWithMasterEntity: self
                  fromRecurrencePattern: &pattern->RecurrencePattern];

So I think this patch is unnecessary and you must put code in `_setupEventRecurrence` method, something like:

    if (pattern == NULL)     [self errorWithFormat: @"OH NO...."];

**Comment by javierag** [comment]
I will redo it using the Blaxter proposal. 

Anyway it seems more safe to do nothing that assume that in setupRecurrenceWithMasterEntity we want to remove the event. setupRecurrenceWithMasterEntity is only called in _setupEventRecurrence so, as longer we  dont add later any call, it should be safe.

**Comment by blaxter** [comment]
:+1: though it would be nice to put oc-calendar prefix on the commit message

**Comment by javierag** [comment]
Added prefix

**Comment by blaxter** [comment]
And now commit message is larger than it should be. The message can be sort saying the same, for instance:

    oc-calendar: Don't change event recurrence if parse failed.

    get_AppointmentRecurrencePattern() can fail to parse the input data, added
    a check to avoid setup the recurrence pattern in that case. This should only
    happen when input data is incorrect.

**Comment by sixstone-qq** [comment]
:+1: to @blaxter comments

**Comment by javierag** [comment]
Commit message changed

**Pull request merged**